sdks/java/nexmark/build.gradle

```diff
@@ -43,8 +68,53 @@ dependencies {
   shadow project(path: ":beam-runners-direct-java", configuration: "shadow")
   shadow library.java.slf4j_jdk14
   testCompile library.java.hamcrest_core
+
+  gradleRun project(path: project.path, configuration: "shadow")
+  gradleRun project(path: nexmarkRunnerDependency, configuration: "shadow")
+
+  // The Spark runner requires the user to provide a Spark dependency. For self-contained
+  // runs with the Spark runner, we can provide such a dependency. This is deliberately phrased
+  // to not hardcode any runner other than :runners:direct-java
+  if (shouldProvideSpark) {
+    gradleRun library.java.spark_streaming
+    gradleRun library.java.spark_core, {
+      exclude group:"org.slf4j", module:"jul-to-slf4j"
+    }
+
+    // TODO: pre-Gradle this was excluded in the SparkRunner overall, in Gradle it is
+    // only excluded in the testRuntimeClasspath.
+    gradleRun project(path: ":beam-sdks-java-core", configuration: "shadow"), {
```

Comment:
You don't need to specify the project, you can just exclude the values configuration wide.